src/data/polynomial/mirror.lean

```diff
@@ -118,9 +118,16 @@ lemma mirror_mirror : p.mirror.mirror = p :=
 polynomial.ext (λ n, by rw [coeff_mirror, coeff_mirror,
   mirror_nat_degree, mirror_nat_trailing_degree, rev_at_invol])
 
+variables {p q}
+
+lemma mirror_eq_iff : p.mirror = q ↔ p = q.mirror :=
+⟨λ h, h ▸ p.mirror_mirror.symm, λ h, h.symm ▸ q.mirror_mirror⟩
+
 lemma mirror_eq_zero : p.mirror = 0 ↔ p = 0 :=
```

Comment:
While you're here, could you try adding `simp` to a few of these lemmas. E.g.,:
```suggestion
@[simp] lemma mirror_eq_zero : p.mirror = 0 ↔ p = 0 :=
```
and maybe also `mirror_trailing_coeff`, `mirror_leading_coeff`?